### PR TITLE
Add handling for cases of file corruption that are not handled by our…

### PR DESCRIPTION
**Pull request by alalal47**
… other methods.

Add option for user to delete file in cases where they are unable to manually rectify corruption by editing the info file(that they messed up)

### CODE REVIEW

File: src/main/java/classify/data/DataReader.java [hunk 1]
```diff
@@ -205,14 +205,60 @@ public static void initialiseData(ArrayList<Student> list,
...
+     * Or, provides users the option to wipe all existing data in the case that they are unable to.
+     */
+    private static void promptDataDeletion() {
+        String input = in.nextLine();
+        if (checkDeletionRequestInput(input)) {
+            DataUI.printDataDeletionFinalConfirmationMessage();
+            input = in.nextLine();
+            if (deletionRequestFinalConfirmation(input)) {
+                DataHandler.deleteStudentInfo();
+            } else {
```
Comment (Cryolian) [inline src/main/java/classify/data/DataReader.java:246]: maybe use returns instead of nested if/else statements for better readability?

**Review by Cryolian — APPROVED** [review]
Logic and code quality seem fine to me.

File: src/main/java/classify/data/DataReader.java [hunk 2]
```diff
@@ -205,14 +205,60 @@ public static void initialiseData(ArrayList<Student> list,
...
+            } catch (ArrayIndexOutOfBoundsException e) {
+                DataUI.printFileCorruptedMessage();
+                DataUI.printDataDeletionPromptMessage();
+                promptDataDeletion();
+                UI.printDivider();
+            } catch (IOException e) {
+                DataUI.printFileCorruptedMessage();
+                DataUI.printDataDeletionPromptMessage();
+                promptDataDeletion();
+                UI.printDivider();
```
Comment (tayponghee) [inline src/main/java/classify/data/DataReader.java:223]: Good use of the UI class to print out messages that corresponds with the corrupted information

File: src/main/java/classify/data/DataWriter.java [hunk 3]
```diff
@@ -62,4 +62,20 @@ public static void createParentFileFolder (String parentPath) {
...
+    public static void emptyDataFile(String folderPath, String filePath) {
+        try {
+            createParentFileFolder(folderPath);
+            FileWriter overwriteBlank = new FileWriter(filePath);
+            overwriteBlank.write("");
+            DataUI.printFileDeletionCompleteMessage();
+        } catch (IOException e) {
+            DataUI.printFileIOError();
+        }
+    }
```
Comment (tayponghee) [inline src/main/java/classify/data/DataWriter.java:80]: Nice addition to allow for file wipe should the file be corrupted and the user chooses to do so

**Review by tayponghee — APPROVED** [review]

**Pull request merged**